### DISCUSSION TITLE
Any known issues with Windows 10 Onedrive?

### DISCUSSION THREAD

**Discussion by Tippon** [abraunegg/onedrive#2208] [Q&A] [answered]
Are there any known issues that affect the Windows 10 version of the official Onedrive client (MS Onedrive for the rest of the question)?

I installed Onedrive on my Mint 20.3 laptop, and manually copied the files before syncing by removing the laptop's secondary drive and installing it into my Windows computer. I told MS Onedrive to keep a local copy of all my files, and copied the folder contents into the laptop's folder. I then replaced the laptop's drive and ran Onedrive, possibly with resync, I can't remember now. From the Mint side, everything seems to have synced successfully, apart from a handful of files that were duplicated and now have the name of the laptop at the end.

Since then, my Windows computer has been running slowly, and I've narrowed the cause down to MS Onedrive. It's constantly trying to download a file, but it doesn't seem to be the same file, and it only gives me the file size. Telling MS Onedrive to pause stops the issue on the computer, restoring everything else back to it's usual speed, until the pause time ends and MS Onedrive starts trying to sync again. I've run a full virus and malware scan just in case, but they found nothing.

Given the timing, I thought I'd check here in case there was a known issue, and hopefully a known fix. Especially since Microsoft, being Microsoft, don't seem to have any logs for their version of Onedrive.

Does anyone know what might be happening please?

**Comment by abraunegg** [comment]
@Tippon 
> Are there any known issues that affect the Windows 10 version of the official Onedrive client (MS Onedrive for the rest of the question)?

There are no known compatibility issues that have been raised in the last 4+ years.

> I installed Onedrive on my Mint 20.3 laptop

What version did you install? What is the output of `onedrive --version` ?

> From the Mint side, everything seems to have synced successfully, apart from a handful of files that were duplicated and now have the name of the laptop at the end.

This happens when a local file exists, a 'newer' file resides on OneDrive and is newer based on timestamp or size, and the 'onedrive' client has no record of the local file, thus, the local file is backed up and renamed and the 'new' file is downloaded.

**Reply by Tippon** [reply]
> What version did you install? What is the output of `onedrive --version` ?

I can't confirm it tonight, but it should be the latest version. I only set Onedrive up on the laptop within the last three weeks.
 
> This happens when a local file exists, a 'newer' file resides on OneDrive and is newer based on timestamp or size, and the 'onedrive' client has no record of the local file, thus, the local file is backed up and renamed and the 'new' file is downloaded.

Sorry, I should have clarified. I knew that could happen, but I added the info just in case there was a known issue and it was related. I'm slightly confused as to why it happened in my case though, as the files hadn't changed. It doesn't matter though, as they've sorted themselves out in the meantime.

> There are no known compatibility issues that have been raised in the last 4+ years.

That's good to know. I thought that was probably the case, but I needed to check, as installing Onedrive on my laptop was the last thing I remember doing before the issue started. I've been on my Windows PC tonight, and MS Onedrive was thrashing my hard drive, and was apparently stuck on downloading a 3.8MB file, but wouldn't tell me which one. I tried unlinking the PC and relinking it, but that did nothing. I've just uninstalled and reinstalled, and am waiting for it to finish processing the existing files to see how it gets on once that's done.
 
It's not thrashing the drive for now, so I'll see how it is in a few hours. Thanks for your help :)

**Comment by Tippon** [comment, marked as answer]
Well that was frustrating!

It turned out to be a possibly corrupted photo that MS Onedrive was trying to download, but wouldn't give me any details on. With a bit of blind luck, I noticed that when I unpaused MS Onedrive, it would say that all my files were in sync, before briefly flashing the name of the file, and quickly changing it to just show the filesize. After a few attempts, I managed to get a screenshot of the name, and tracked it down.

I found three copies, so moved them to a temp folder and confirmed that one was displaying properly. I deleted the copies from my Windows Onedrive folder, and put the known good copy back, and everything seems to be working again.

You'll be happy to know that through all this, your Onedrive has been performing perfectly. Something that you're doing stopped the client from hanging when it got to the corrupted photo, and let everything sync to my laptop as it should.

**Comment by abraunegg** [comment]
@Tippon 
Glad you got this sorted.

> I can't confirm it tonight, but it should be the latest version. I only set Onedrive up on the laptop within the last three weeks.

Please make sure as you are using Mint 20.3 - you install the application *correctly* following https://github.com/abraunegg/onedrive/blob/master/docs/ubuntu-package-install.md as this will ensure that your application is kept up-to-date and current.
